--- /usr/lib/fai/get-config-dir.orig	2007-09-21 10:13:24.000000000 +0200
+++ /usr/lib/fai/get-config-dir	2007-09-21 10:14:11.000000000 +0200
@@ -9,6 +9,11 @@
 # Short-Description: get $FAI directory from $FAI_CONFIG_SRC
 ### END SUBROUTINE INFO
 
+if [ -n "FAI_NO_CONFIG_SRC" ] ; then
+	echo "Notice: skipping get-dir-config as requested"
+	exit 0
+fi
+
 if [ -z "$FAI_CONFIG_SRC" ]; then
 	sendmon "TASKERROR get_fai_dir 21"
 	echo "Error: Provide the URL to obtain the fai config storage in \$FAI_CONFIG_SRC"
