--- /usr/lib/fai/subroutines-linux	2007-10-17 21:21:11.000000000 +0000
+++ /usr/lib/fai/subroutines-linux	2007-10-17 21:21:03.000000000 +0000
@@ -124,7 +124,7 @@
     
     local dversion=$(dpkg -l debootstrap | grep debootstrap | cut -f7 -d' ')
     echo "Creating base system using debootstrap version $dversion"
-    echo "Calling debootstrap $1 $FAI_ROOT $2"
+    echo "Calling debootstrap $1 $FAI_DEBOOTSTRAP_OPTS $FAI_ROOT $2"
     yes '' | LC_ALL=C debootstrap $FAI_DEBOOTSTRAP_OPTS $1 $FAI_ROOT $2
 }
 # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
