--- /usr/lib/fai/subroutines.orig	2007-09-21 10:46:23.000000000 +0200
+++ /usr/lib/fai/subroutines	2007-09-21 19:36:34.000000000 +0200
@@ -476,6 +476,9 @@
     task prepareapt
     task updatebase
     task instsoft
+    if grep -q 'E: Sub-process /usr/bin/dpkg returned an error code' $LOGDIR/software.log ; then
+	    die "Error in software installation task."
+    fi
     task configure
     task finish
 
